### PR TITLE
[development] History - Allow "First N" query

### PR DESCRIPTION
**Pull request by thiagodpf**
* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

feature

* **What is the current behavior?** (You can also link to an open issue here)

It is only possible to sort by the last data registered in the bank

* **What is the new behavior (if this is a feature change)?**

It is possible to sort by the first data registered in the bank (the oldest ones)

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

no

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)

dojot/dojot#1701

* **Other information**:

### CODE REVIEW

**Comment by mprevide** [comment]
you need add the indexs too

**Comment by mprevide** [comment]
the base should be v0.4.3, but you can copy your changes

File: history/api/models.py [hunk 1]
```diff
@@ -117,7 +117,17 @@ def parse_request(request, attr):
         """ returns mongo compatible query object, based on the query params provided """
         logger.debug('DeviceHistory.parse_request [start]')
 
-        if 'lastN' in request.params.keys():
+        limit_val = False
+        sort = [('ts', pymongo.DESCENDING)]
+
+        if 'firstN' in request.params.keys():
+            try:
+                limit_val = int(request.params['firstN'])
```
Comment (mprevide) [inline history/api/models.py:125]: I think that's the idea. Maybe I just need to confirm with who is asking.

File: docs/history.apib [hunk 2]
```diff
@@ -140,30 +144,30 @@ If no query string is set, then all values from all attributes are returned.
...
                         "payload": "json"
                     },
-                    "value": "1013",
+                    "value": "1053",
                     "attr": "pressure"
                   },
                   {
                     "device_id": "b374a5",
-                    "ts": "2018-02-14T13:42:33.306000Z",
+                    "ts": "2010-01-01T10:12:22.306000Z",
```
Comment (mprevide) [inline docs/history.apib:170]: Por que 2010?

Comment (thiagodpf) [reply docs/history.apib:170]: The idea was to bring the oldest records, but I just saw that I forgot to change the parameter `dateFrom=2010-01-01T00:00:00Z` as well :sweat_smile:

**Review by mprevide — APPROVED** [review]

**Pull request merged**